### PR TITLE
Dces-467 Add the ability to run report monthly as well as daily

### PR DESCRIPTION
**Pull request by vvartakMoJ**
[Link to story](https://dsdmoj.atlassian.net/browse/DCES-467)

1. Added a new parameter of ReportTitle (e.g. Monthly/Daily/AdHoc) to endpoints and shell scripts (used to be hard-coded as Monthly)
2. Added CRON schedule for the daily reports to the yaml files.
3. New enums Report Type (Contribution or FDC) and ReportPeriod (Monthly or Daily)
4. If there is no data for the period (e.g. single day) then no exception is thrown. Instead, an empty CSV file is produced with a message inside saying that there is no data for the given period.
5. Added Feature Flags to enable or disable the daily report from a secret (created in Amazon Secret Manager, available as a K8s secret)
6. If the daily report generation is disabled (via the feature flag) then no email is sent but a message is logged to that effect, at the scheduled time.
 
## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/DcesReportFileService.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
 public interface DcesReportFileService {
     List<String> getFiles(LocalDate start, LocalDate end);
 
-    File processFiles(List<String> files, LocalDate start, LocalDate finish)
+    File processFiles(List<String> files, String reportTitle, LocalDate start, LocalDate finish)
```
[the remark this answers is not in the document]
Comment (aminur-rouf) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/DcesReportFileService.java:18]: Have a read of this and then make an informed decision on whether or not to pass an object . 
https://refactoring.guru/smells/long-parameter-list
There's no right or wrong answer to this, so I would be tempted to leave the code as is and decide next time.

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileService.java [hunk 2]
```diff
@@ -29,65 +29,70 @@ public class CSVFileService {
...
-    public void writeFdcToCsv(FdcFile fdcFile, File targetFile, LocalDate fromDate, LocalDate toDate) throws IOException {
-        List<Fdc> fdcList = fdcFile.getFdcList().getFdc();
-        String dateGenerated = DateUtils.convertXmlGregorianToString(fdcFile.getHeader().getDateGenerated());
         // file-writer initialise
         try (FileWriter fw = new FileWriter(targetFile, true)) {
-            if (targetFile.length() == 0) {
-                writeFdcHeader(fw, fromDate, toDate);
+            writeFdcHeader(fw, reportTitle, fromDate, toDate);
+            boolean someDataFound = false;
+            for (FdcFile fdcFile : fdcFiles) {
```
[the remark this answers is not in the document]
Comment (vvartakMoJ) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileService.java:77]: The need to calculate dateGenerated in the first loop means that the result will be as unreadable, if not more, in my opinion. So leaving it as it is.

File: dces-report-service/src/test/java/uk/gov/justice/laa/crime/dces/report/service/FdcFileServiceConnectionTest.java [hunk 3]
```diff
@@ -23,9 +23,9 @@
 class FdcFileServiceConnectionTest {
     @InjectSoftAssertions
     private SoftAssertions softly;
-    private static final LocalDate startPeriod = LocalDate.of(2021, 1, 2);
+    private static final LocalDate startPeriod = LocalDate.of(2024, 10, 8);
 
-    private static final LocalDate finishPeriod = LocalDate.of(2021, 1, 26);
+    private static final LocalDate finishPeriod = LocalDate.of(2024, 10, 10);
```
Comment (LAA-Dave-Edwards) [inline dces-report-service/src/test/java/uk/gov/justice/laa/crime/dces/report/service/FdcFileServiceConnectionTest.java:28]: Will the date cause any issues as time goes on?

**Review by LAA-Dave-Edwards — APPROVED** [review]
Looks good to me.

File: dces-report-service/src/test/java/uk/gov/justice/laa/crime/dces/report/service/FdcFileServiceConnectionTest.java (hunk 3 above)
Comment (vvartakMoJ) [reply dces-report-service/src/test/java/uk/gov/justice/laa/crime/dces/report/service/FdcFileServiceConnectionTest.java:28]: I don't think so, I don't think there was really a need to change it here either.

**Pull request merged**